### PR TITLE
Corrected handling of Unit

### PR DESCRIPTION
**Pull request by mattgreenuk**

### CODE REVIEW

File: app/uk/gov/hmrc/apiplatformmicroservice/apidefinition/connectors/ApiDefinitionConnector.scala [hunk 1]
```diff
@@ -34,17 +34,11 @@ trait ApiDefinitionConnector extends ApiDefinitionConnectorUtils with ApiDefinit
...
-        case _: NotFoundException =>
-          Logger.info("Not found")
-          List.empty
-        case NonFatal(e)          =>
-          Logger.error(s"Failed $e")
-          throw e
-      }
+    http.GET[Option[List[APIDefinition]]](definitionsUrl(serviceBaseUrl), Seq("type" -> "all"))
+    .map(_ match {
+      case None => List.empty
```
Comment (peteslater-ee) [inline app/uk/gov/hmrc/apiplatformmicroservice/apidefinition/connectors/ApiDefinitionConnector.scala:39]: Do we bother typing the empty lists?

Comment (AndySpaven) [reply app/uk/gov/hmrc/apiplatformmicroservice/apidefinition/connectors/ApiDefinitionConnector.scala:39]: No

File: build.sbt [hunk 2]
```diff
@@ -9,7 +9,7 @@ lazy val scoverageSettings = {
   import scoverage.ScoverageKeys
   Seq(
     // Semicolon-separated list of regexs matching classes to exclude
-    ScoverageKeys.coverageExcludedPackages := ";.*\\.domain\\.models\\..*;uk\\.gov\\.hmrc\\.BuildInfo;.*\\.Routes;.*\\.RoutesPrefix;.*Filters?;MicroserviceAuditConnector;Module;GraphiteStartUp;.*\\.Reverse[^.]*",
+    ScoverageKeys.coverageExcludedPackages := ";.*\\.domain\\.models\\..*;uk\\.gov\\.hmrc\\.BuildInfo;.*\\.Routes;.*\\.RoutesPrefix;.*Filters?;MicroserviceAuditConnector;Module;GraphiteStartUp;.*\\.Reverse[^.]*;uk\\.gov\\.hmrc\\.apiplatformmicroservice\\.apidefinition\\.controllers\\.binders",
```
Comment (peteslater-ee) [inline build.sbt:12]: Do we prefer this way of doing exclusions, with all the backslashes, or do we prefer the way we do it in other repos?

https://github.com/hmrc/third-party-developer-frontend/blob/master/build.sbt

Comment (AndySpaven) [reply build.sbt:12]: Not going to change this in this commit.  build tidyup is a whole new issue.

**Review by peteslater-ee — APPROVED** [review]

**Pull request merged**